server/src/main/java/com/cloud/network/NetworkModelImpl.java

```diff
@@ -2723,6 +2733,12 @@ public Pair<String, String> getNetworkIp6Dns(final Network network, final DataCe
         if (StringUtils.isNotBlank(network.getIp6Dns1())) {
             return new Pair<>(network.getIp6Dns1(), network.getIp6Dns2());
         }
+        if (network.getVpcId() != null) {
+            Vpc vpc = vpcDao.findById(network.getVpcId());
+            if (vpc != null && StringUtils.isNotBlank(vpc.getIp6Dns1())) {
+                return new Pair<>(vpc.getIp6Dns1(), vpc.getIp6Dns2());
+            }
+        }
```

Comment:
this block is implemented twice. can you extract to a method?

Reply:
@DaanHoogland 
There is a small difference:
`ip4` vs `ip6`